allennlp/nn/beam_search.py

```diff
@@ -381,3 +409,178 @@ def _search(
         all_predictions = torch.cat(list(reversed(reconstructed_predictions)), 2)
 
         return all_predictions, last_log_probabilities
+
+    @classmethod
+    def without_sampling(
+        cls,
+        end_index: int,
+        max_steps: int = 50,
+        beam_size: int = 10,
+        per_node_beam_size: int = None,
+        sampler: Sampler = None,
+    ) -> "BeamSearch":
+        """
+        Given an index of the end token in target vocabulary, return a BeamSearch object
+        that can be used to find `beam_size` candidate sequences, found by sampling from
+        the top `k` tokens to choose from at each step.
+
+        # Parameters
+
+        end_index : `int`
+            The index of the "stop" or "end" token in the target vocabulary.
+        max_steps : `int`, optional (default = `50`)
+            The maximum number of decoding steps to take, i.e. the maximum length
+            of the predicted sequences.
+        beam_size : `int`, optional (default = `10`)
+            The width of the beam used.
+        per_node_beam_size : `int`, optional (default = `beam_size`)
+            The maximum number of candidates to consider per node, at each step in the search.
+            If not given, this just defaults to `beam_size`. Setting this parameter
+            to a number smaller than `beam_size` may give better results, as it can introduce
+            more diversity into the search. See
+            [*Beam Search Strategies for Neural Machine Translation*, Freitag and Al-Onaizan, 2017]
+            (https://arxiv.org/abs/1702.01806).
+        sampler : `Sampler`, optional (default = `None`)
+            A sampler that can be used to select subsequent tokens.
+            If not given, search defaults to selecting the top `per_node_beam_size` tokens at each
+            step
+
+        # Returns
+
+        `BeamSearch` object with the defined parameters
```

Comment:
I think it's okay to omit parameter descriptions here since they're given above.
```suggestion
```